tket2-py/src/optimiser.rs

```diff
@@ -61,6 +61,12 @@ impl PyBadgerOptimiser {
     ///     If `None` the optimiser will run indefinitely, or until `timeout` is
     ///     reached.
     ///
+    /// * `max_circuit_cnt`: The maximum number of circuits to process before
+    ///     stopping the optimisation.
```

Comment:
Mention that it doesn't apply to multithreaded badger. Maybe log a warning in `badger_multithreaded` if `max_circuit_cnt.is_some()` ?